apps/openssl.c

```diff
@@ -316,6 +316,37 @@ static void list_missing_help(void)
     }
 }
 
+static void list_objects(void)
+{
+    int max_nid = OBJ_new_nid(0);
+    int i;
+
+    /* Skip 0, since that's NID_undef */
+    for (i = 1; i < max_nid; i++) {
+        const ASN1_OBJECT *obj = OBJ_nid2obj(i);
+        char oid_buf[128];       /* I dare you to find such a long OID */
```

Comment:
A challenge is a bad thing to pose a bored programmer.

Reply:
I think you'll find that `OBJ_create` call quite a challenge in itself ;-)

Reply:
But sure, someone might do `OBJ_create("2.1.20.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999.999999", "myShortName", "My Longer Name!");`

Reply:
The problem here is that `OBJ_obj2txt` will silently stop filling the buffer when its size has been reached, so while a limited buffer is without harm, the OID output may be truncated, possibly invisibly.  Fixing that is a matter for a different PR, methinks.

Reply:
Oh!  No, I was mistaken...  `OBJ_obj2txt` returns the size the buffer *should* have.  Ok, will change the buffer to be heap-reallocatable.